File: datastruct-algoritem-java/src/main/java/datastruct/queue/QueueTest.java
package com.datastruct.queue;

/**
 * @author czx
 * @date 2019/12/24
 */
public class QueueTest {
    public static void main(String[] args) {
        Queue<Integer> q = new Queue<>();
        for (int i = 0; i < 10; i++){
            q.offer(i);
        }

        q.forEach(e -> System.out.print(e + " "));
        System.out.println();
        System.out.println("q.size() = " + q.size());
        Integer pop = q.pop();
        System.out.println("pop = " + pop);
        q.pop();
        q.forEach(e -> System.out.print(e + " "));
        System.out.println("q.size() = " + q.size());
    }
}

File: datastruxt-algoritem-go/datastruct/queue/queue_test.go
package queue

import (
	"fmt"
	"testing"
)

// Create by czx on 2019/12/25

func TestQueue(t *testing.T) {
	q := NewQueue()
	for i := 0; i < 10; i++ {
		q.Push(i)
	}
	q.Foreach(func(v interface{}) {
		fmt.Printf("%v ", v)
	})
	fmt.Println()
	fmt.Println("size: ", q.Size())
	peek := q.Peek()
	fmt.Println("peek value: ", peek)
	if ok, i := q.Pop(); ok {
		fmt.Println("pop success: ", i)
	}
	if ok, i := q.Pop(); ok {
		fmt.Println("pop success: ", i)
	}
	q.Clear()
	fmt.Println("queue is empty: ", q.Empty())
}
